sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -257,10 +257,19 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
                 planLater(child))
             }
           } else if (functionsWithDistinct.isEmpty) {
+            // Check if the child operator satisfies the group-by distribution requirements
```

Comment:
Why not move this code block into `aggregate.Utils.planAggregateWithoutDistinct`?